### PR TITLE
Picodet improvements is

### PR DESCRIPTION
**Pull request by IgorSusmelj**
## What has changed and why?

This PR is a focused cleanup and optimization pass for our PicoDet model family.
Development was guided by real deployment constraints on a Jetson Orin Nano (8GB), with architecture changes aimed at the best speed/accuracy trade-off on that device.

The models now support NMS-free inference using a YOLOv10-style one-to-one (O2O) head.
We still train with two heads in parallel:
- Dense one-to-many head: provides rich supervision (more positive assignments), stabilizes optimization, and improves recall/localization quality during training.
- O2O head: learns unique prediction assignment and is used for final NMS-free inference.

In short, the dense head helps the model learn better, while the O2O head gives fast, clean, NMS-free predictions at inference time.

- Model config changes
    - remove picodet/l-416 preset and keep only picodet/l-640 for L
    - update auto transform resolution logic so L defaults to 640x640
- Architecture changes 
  - new O2O head for NMS free detection
  - replace SwiGLU with ReLU activations
  - remove unfriendly ops for old hardware (topk sampling)
- Inference/export changes
    - inference and ONNX/TensorRT export now use O2O decoding path (NMS-free)
- Initialization/fine-tuning improvements
    - add optional backbone_weights loading utility
    - keep optional backbone_freeze flow for transfer learning setups

Outstanding Tasks

- Add docs for picodet --> follow-up issue
- Upload new checkpoints to storage

Model size
| Model Name | Old Arch Parameters  | New Arch Parameters |
|-------------|----|------|
|picodet-s| 1.17M |1.325M|
|picodet-l| 3.75M |3.5M|

Runtime numbers for Jetson Orin Nano, with tensorrt 10.3.0. CUDA 12.6, JetPack 6.2.1:
- note that the old architecture was measured without NMS while the new architecture no longer needs nms
  
|     Model      | Input Res.      | Inference Engine | Old Arch. time | New Arch. time | Old Arch mAP50..95 | New Arch mAP50..95 |
|-----------|-----------------|------------------|----------------|----------------|--------------------|--------------------|
| s | onnx, fp32, cpu | 416x416          | 115ms          | 51.6ms         | 26.7                  | 27.0                  |
| s | onnx, fp32, gpu | 416x416          | 33.1ms         | 23.5ms         | 26.7                  | 27.0                  |
| s | tensor-rt, fp32 | 416x416          | -              | 8.41ms         |26.7                  | 27.0                  |
| s | onnx, fp16, cpu | 416x416          | 115ms          | 51.6ms         | 26.7                  | 27.0                  |
| s | onnx, fp16, gpu | 416x416          | 42.7ms         | 18.9ms         | 26.7                  | 27.0                  |
| s | tensor-rt, fp16 | 416x416          | 9.82ms              | 5.84ms         |26.7                  | 27.0                  |
| l | onnx, fp32, cpu | 416x416          | 182ms          | -         | 32.0                  | -                  |
| l | onnx, fp32, gpu | 416x416          | 38.8ms         | -         | 32.0                  | -                  |
| l | onnx, fp32, cpu | 640x640          | -          | 173ms         | -                  | 35.0                  |
| l | onnx, fp32, gpu | 640x640          | -         | 46.7ms         | -                  | 35.0                  |
| l | tensor-rt, fp32 | 640x640          | -         | 25.4ms         | -                  | 35.0                  |
| l | onnx, fp16, cpu | 416x416          | 120ms          | -         | 32.0                  | -                  |
| l | onnx, fp16, gpu | 416x416          | 43.7ms         | -         | 32.0                  | -                  |
| l | tensor-rt, fp16 | 416x416          | 12.51ms         | -         | 32.0                  | -                  |
| l | onnx, fp16, cpu | 640x640          | -          | 178ms         | -                  | 35.0                  |
| l | onnx, fp16, gpu | 640x640          | -         | 42.3ms        | -                  | 35.0                  |
| l | tensor-rt, fp16 | 640x640          | -         | 12.5ms         | -                  | 35.0                  |

## How has it been tested?

- Results above have been obtained using the newly trained checkpoints

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [X] Yes
- [ ] Not needed (internal change)

## Did you update the documentation?

- [ ] Yes
- [ ] Not needed (internal change without effects for user)

### CODE REVIEW

File: src/lightly_train/_task_models/picodet_object_detection/pico_head.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def forward(self, x: Tensor) -> Tensor:
         x = F.softmax(x, dim=-1)
 
         # Compute expectation
-        project: Tensor = self.project  # type: ignore[assignment]
+        project: Tensor = self.project.to(dtype=x.dtype, device=x.device)  # type: ignore[assignment]
```
Comment (guarin) [inline src/lightly_train/_task_models/picodet_object_detection/pico_head.py]: The `device=x.device` is not needed. I agree with the copilot review below that I would convert the buffer only once:
```
if self.project.dtype != x.dtype:
    self.project = self.project.to(dtype=x.dtype)
```

File: src/lightly_train/_task_models/picodet_object_detection/task_model.py [hunk 2]
```diff
@@ -167,6 +191,106 @@ def __init__(
...
+            keep = (scores >= threshold) & (scores == pooled)
+            total_peaks[level_idx] = keep.sum()
+        batch_size = cls_scores_list[0].shape[0]
+        return total_peaks / float(batch_size)
+
+    def _add_onnx_metadata(self, out: PathLike) -> None:
+        """Attach class mapping metadata to exported ONNX model."""
+        if self.classes is None:
+            return
+        try:
```
Comment (guarin) [inline src/lightly_train/_task_models/picodet_object_detection/task_model.py:228]: I believe you should be able to add metadata already during the onnx export and not reload the model

File: src/lightly_train/_task_models/picodet_object_detection/task_model.py [hunk 3]
```diff
@@ -167,6 +191,106 @@ def __init__(
...
+
+    def load_backbone_weights(self, path: PathLike) -> None:
+        """Load backbone weights from a checkpoint file.
+
+        Args:
+            path: Path to a .pt file (e.g., exported_last.pt).
+        """
+        if not os.path.exists(path):
+            logger.error("Checkpoint file not found: %s", path)
+            return
```
Comment (guarin) [inline src/lightly_train/_task_models/picodet_object_detection/task_model.py]: Use same code as everywhere else in the codebase:

```suggestion
        path = Path(path).resolve()
        if not path.exists():
            raise FileNotFoundError(f"Backbone weights file not found: '{path}'")
```

File: src/lightly_train/_task_models/picodet_object_detection/task_model.py (hunk 2 above)
Comment (IgorSusmelj) [reply src/lightly_train/_task_models/picodet_object_detection/task_model.py:228]: Not possible. There are issues and proposals. The common workaround is saving --> loading & modifying --> saving

Issue: https://github.com/pytorch/pytorch/issues/42808
PR that did not get merge: https://github.com/isaac-sim/IsaacLab/issues/998
